src/main/eo/org/eolang/collections/list.eo

```diff
@@ -49,23 +49,37 @@
   # for the index and the third one for the element
   # of the array.
   [a f] > reducedi
-    rec-reduced (* a 0) f arr > @
+    if. > @
+      eq.
+        0
+        length.
+          arr
+      a
+      rec-reduced
+        (* a 0)
+        f
+        arr
 
     [acc-index func carr] > rec-reduced
       (acc-index.at 0) > acc
       (acc-index.at 1) > index
-      if. > @
-        (index.lt (carr.length))
-        rec-reduced
-          *
+      seq > @
+        func > new-acc!
+          acc
+          index
+          (carr.at index)
+        if.
+          ((index.plus 1).eq (carr.length))
+          new-acc
+          rec-reduced
+            *
+              func
+                acc
+                index
+                (carr.at index)
+              (index.plus 1)
```

Comment:
Here we done 2 equal things two times: we accumulate `func` before `if`, and after that again when there is a recursive call. But I don't really understand here, what is really changed in logic of this object, comparing to old version?
In old version If we can't apply `rec-reduced` one more time, we returned the last result - `acc`, otherwise we called it again.
Aren't we doing the same thing now after the changes?

Reply:
@MikhailLipanin first, I would like to say that there was a bug (which is noted in the attached ticket) and I solved it. It seems to me that the person who solves the bug shouldn't explain why the last version of the code didn't work if the problem is solved now.
@yegor256 what do you think about it, maybe I'm wrong?

Reply:
@MikhailLipanin However, I will point out that indeed the code seems similar. And to notice the difference you need to seriously think about what really will be the order of datumization. It is important to understand that EOLANG is not a serial language, and that is why these versions of the code are different.